### PR TITLE
fix: a11y: search component submit accessible naming

### PR DESCRIPTION
**Pull request by shkeating**
# Summary

adds support for accessible name for search submit button in small search variant, and enhances naming capabilities on all search variants submit button to prevent duplicate naming issues on pages with multiple search components. Example of this scenario in USSF portal provided in screen shots.

## Related Issues or PRs

- [Space Force Portal repo small search accessibility branch](https://github.com/USSF-ORBIT/ussf-portal-client/pull/1205), needs accessible name for button, and requires it be unique so the button name does not clash with site search field in header.

## How To Test

check output in ANDI, which should reflect the fallback name of "Search". Note, the aria label added to the non-small variants will not conflict with the typed name button text in the default instances, because the aria label will override the button text in the accessibility tree. This will allow us to have "search" appear visually for users who will have visual context, but more precise labeling for users of assistive tech.

I encourage reviewers to be vigilant in the changes made to the Search test file. My intention is to pull this into the USSF portal repo to ensure it works as expected.

#### Example of multiple searches in a single interface

image

##### Buttons with inner text

image

##### Small Search: Buttons without Inner Text

image

### CODE REVIEW

**Comment by AnnaGingle** [comment]
Taking a look now!

**Review by AnnaGingle — DISMISSED** [review]
The ANDI output works as expected. No visual changes in the design. Great work!

File: src/components/Search/SearchButton/SearchButton.tsx [hunk 1]
```diff
@@ -13,12 +13,14 @@ type SearchButtonProps = {
...
+  buttonAriaLabel?: string
 }
 
 export const SearchButton = ({
   size,
   className,
   i18n,
+  buttonAriaLabel,
 }: SearchButtonProps): React.ReactElement => {
   const buttonText = i18n?.buttonText || 'Search'
```
Comment (brandonlenz) [inline src/components/Search/SearchButton/SearchButton.tsx:25]:
```suggestion
  const buttonText = i18n?.buttonText || 'Search'
  const ariaLabel = buttonAriaLabel || buttonText
```

File: src/components/Search/SearchButton/SearchButton.tsx [hunk 2]
```diff
@@ -35,12 +37,17 @@ export const SearchButton = ({
     <div className={classes}>
       <Button type="submit">
         {!isSmall && (
-          <span className="usa-search__submit-text">{buttonText}</span>
+          <span
+            className="usa-search__submit-text"
+            aria-label={buttonAriaLabel || buttonText}>
```
Comment (brandonlenz) [inline src/components/Search/SearchButton/SearchButton.tsx:42]:
```suggestion
            aria-label={ariaLabel}>
```

File: src/components/Search/SearchButton/SearchButton.tsx [hunk 3]
```diff
@@ -35,12 +37,17 @@ export const SearchButton = ({
...
+            {buttonText}
+          </span>
         )}
         <Icon.Search
           className="usa-search__submit-icon"
           name={buttonText}
           size={3}
+          aria-label={buttonAriaLabel || buttonText}
```
Comment (brandonlenz) [inline src/components/Search/SearchButton/SearchButton.tsx:50]:
```suggestion
          aria-label={ariaLabel}
```

**Review by brandonlenz — APPROVED** [review]
Non blocking, but minor code organizational nit recommended in review comments

**Comment by werdnanoslen** [comment]
@shkeating is the tooltip supposed to be always visible, as Happo's showing?

**Comment by shkeating** [comment]
status update: not behaving as expected. aria-label attribute getting added with incorrect name to form element insead of the search button. 

image

**Review by gidjin — APPROVED** [review]
I used yarn link as described in our docs [here](https://github.com/trussworks/react-uswds/blob/main/docs/contributing.md#yarn-link) to import this branch into our project and the aria-label shows up for search buttons as expected when I run it all locally.

SCR-20240221-s0n

SCR-20240221-s1b

**Pull request merged**